src/Lecture3.hs

```diff
@@ -137,10 +155,10 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
-
+  List1 a as <> List1 b bs = List1 a (as <> [b] <> bs)
```

Comment:
You can slightly improve the performance here by using the `:` operator instead of list append `<>` or `++` 🐎 
```suggestion
  List1 a as <> List1 b bs = List1 a (as <> (b : bs))
```

Reply:
Yes, true! Don't know, still not so comfortable with `:`, think it will come at some point.